### PR TITLE
View helper methods

### PR DESCRIPTION
**Pull request by gardnerapp**
If/Else conditional statements based on a users session status are relatively common at the controller and view level of applications. These conditional statements are repetitive, accumulate quickly and make our code bulky . Great programs are DRY and this pull request seeks to DRY up our code by creating helper methods that abstract if/else conditionals into methods where a block is called based on a resources sign_in/out state. 

Examples: 

View -> 

`user_signed_out do
   <%= link_to "Sign In", new_user_session_path %>
end`

Controller ->

`
def some_action;
   admin_signed_in { render @posts };
   admin_signed_out { redirect_to root_path };
end 
`

### CODE REVIEW

File: test/views/helper_methods_test.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+require 'test_helper'
+
+module ActionView
+  include Devise::Views::Helpers
+end
+
+class ViewsHelperMethodsTest < Devise::IntegrationTest
+
+  test 'Action View includes Devise::Views::Helpers' do
+    assert_includes  ActionView.ancestors, Devise::Views::Helpers
```
Comment (marivaldo) [inline test/views/helper_methods_test.rb:11]:
```suggestion
    assert_includes ActionView.ancestors, Devise::Views::Helpers
```

Comment (gardnerapp) [reply test/views/helper_methods_test.rb:11]: Can we get this pulled?

Comment (gardnerapp) [reply test/views/helper_methods_test.rb:11]: Thank you for the review. Could we get this pull request submitted?

**Comment by nashby** [comment]
Hey @gardnerapp! Thanks for the PR but I think it's something that you can have in your application as a helper

**Pull request closed without merging**